controllers/activegate/capability/capability.go

```diff
@@ -0,0 +1,76 @@
+package capability
+
+import (
+	dynatracev1alpha1 "github.com/Dynatrace/dynatrace-operator/api/v1alpha1"
+)
+
+type Configuration struct {
+	SetDnsEntryPoint     bool
+	SetReadinessPort     bool
+	SetCommunicationPort bool
+	CreateService        bool
+	ServiceAccountOwner  string
+}
+
+type Capability struct {
+	ModuleName     string
+	CapabilityName string
+	Properties     *dynatracev1alpha1.CapabilityProperties
+	Configuration
+}
+
+func (c *Capability) CalculateStatefulSetName(instanceName string) string {
+	return instanceName + "-" + c.ModuleName
+}
+
+type CapabilityType int
+
+const (
+	KubeMon CapabilityType = iota
+	Routing
+	Metrics
+)
+
+func NewCapability(ct CapabilityType, crProperties *dynatracev1alpha1.CapabilityProperties) *Capability {
+	if crProperties == nil {
+		return nil
+	}
+
+	switch ct {
+	case KubeMon:
+		return &Capability{
+			ModuleName:     "kubemon",
+			CapabilityName: "kubernetes_monitoring",
+			Properties:     crProperties,
+			Configuration: Configuration{
+				ServiceAccountOwner: "kubernetes-monitoring",
+			},
+		}
+	case Routing:
+		return &Capability{
+			ModuleName:     "routing",
+			CapabilityName: "MSGrouter",
+			Properties:     crProperties,
+			Configuration: Configuration{
+				SetDnsEntryPoint:     true,
+				SetReadinessPort:     true,
+				SetCommunicationPort: true,
+				CreateService:        true,
+			},
+		}
+	case Metrics:
+		return &Capability{
+			ModuleName:     "metrics",
+			CapabilityName: "metrics_ingest",
+			Properties:     crProperties,
+			Configuration: Configuration{
+				SetDnsEntryPoint:     true,
+				SetReadinessPort:     true,
+				SetCommunicationPort: true,
+				CreateService:        true,
+			},
+		}
+	}
+
+	return nil
```

Comment:
To improve resiliency, this function could be split into `NewKubeMonCapability`, `NewRoutingCapability` and `NewMetricsCapability`. This would remove the necessity for returning `nil` if the type is not supported and consecutive validation checks.